Fixed cluttered buttons in Theming Control Panel (Issue #212)

rohnsha0 opened the pull request:
as said in the issue #212

mister-roboto commented:
@rohnsha0 thanks for creating this Pull Request and helping to improve Plone!

TL;DR: Finish pushing changes, pass all other checks, then paste a comment:
```
@jenkins-plone-org please run jobs
```

To ensure that these changes do not break other parts of Plone, the Plone test suite matrix needs to pass, but it takes 30-60 min.  Other CI checks are usually much faster and the Plone Jenkins resources are limited, so when done pushing changes and all other checks pass either [start all Jenkins PR jobs yourself](https://jenkinsploneorg.readthedocs.io/en/latest/run-pull-request-jobs.html#run-a-pull-request-job), or simply add the comment above in this PR to start all the jobs automatically.

Happy hacking!

rohnsha0 commented:
@jenkins-plone-org please run jobs

stevepiercy commented:
This PR does not resolve the issue. Please see this comment: https://github.com/plone/Products.CMFPlone/issues/3595#issuecomment-1219026687

rohnsha0 commented:
> This PR does not resolve the issue. Please see this comment: [plone/Products.CMFPlone#3595 (comment)](https://github.com/plone/Products.CMFPlone/issues/3595#issuecomment-1219026687)

@stevepiercy changing the div class doesn't make any changing.... that just changes the top margin....

rohnsha0 commented:
![image](https://user-images.githubusercontent.com/86848116/186178418-a8859469-4e93-44ed-850a-9b6c36d47b90.png)
LEFT: `<div class="btn-group">`
RIGHT: `<div class="m-1">`

rohnsha0 commented:
![image](https://user-images.githubusercontent.com/86848116/186179082-925b3cb1-5406-4fb3-a7d5-109398f09126.png)
here's the inspector...

rohnsha0 commented:
![image](https://user-images.githubusercontent.com/86848116/186180323-9a01b8a3-f922-4ea1-a0d0-ab54517d482b.png)
After changes in this PR! @stevepiercy

jensens commented:
Thanks for touching this! Idea:
At the moment all are blue (primary). But by convention Upload should be green (it creates something). Test and help could be secondary. Also below buttons could be better. Deactivate could be red, clear cache yellow. All small size and a button group might look better too.

rohnsha0 commented:
@jensens okay... I'll try to perform changes as you suggested... 
But I had a question... Our PLONE control panel to date doesn't have many colors...
Implementing each button with some colorful approach... would that look good?

jensens commented:
> @jensens okay... I'll try to perform changes as you suggested... But I had a question... Our PLONE control panel to date doesn't have many colors... Implementing each button with some colorful approach... would that look good?

We have green at least for creation, should be visible e.g. in users and groups.

rohnsha0 commented:
> 

okay...
Pushing soon...

rohnsha0 commented:
![image](https://user-images.githubusercontent.com/86848116/186211154-70e0f1b6-225d-43bb-ad0f-809f18cce865.png)

@jensens now?

stevepiercy commented:
> > This PR does not resolve the issue. Please see this comment: [plone/Products.CMFPlone#3595 (comment)](https://github.com/plone/Products.CMFPlone/issues/3595#issuecomment-1219026687)
> 
> @stevepiercy changing the div class doesn't make any changing.... that just changes the top margin....

The original issue (https://github.com/plone/Products.CMFPlone/issues/3595) says:

> The top 3 buttons have no paddings, the placement of the bottom buttons also needs some love.

You are focusing on the bottom buttons, and ignoring the top buttons. If you apply the change I suggested to the top buttons, it should be fine.

Your target should be to replicate the existing look and feel of the Plone 5.2 UI, which is called Classic UI in Plone 6.

## Screen shot of Theming control panel on Plone 5.2 demo.

https://demo.plone.org/@@theming-controlpanel

![Screen Shot 2022-08-23 at 2 15 20 PM](https://user-images.githubusercontent.com/102112/186267041-5b54a82e-d1fb-463a-a26f-358659d3c228.png)

@jensens where is the style guide for the Plone Classic UI? All those color changes to the buttons seems inconsistent to the entire Classic UI. IMO it also looks bad.

rohnsha0 commented:
I got your point @stevepiercy 
Implementing the suggested changes in a while!

rohnsha0 commented:
I too thought that v6b1's theming panel should have the same functionality as v5.2.9
But nevertheless I'll add the required buttons to v6 control panels

stevepiercy commented:
Please do not _add_ buttons, and limit changes to _spacing_ the _existing_ buttons, as the original issue notes.

If you would like to add functionality with new buttons, then that should be discussed either:

1. In a new and separate issue together with the Classic UI team.
2. In the [Community Forum > Developer > User Interface category](https://community.plone.org/c/development/user-interface/14).
3. In [Discord, #classic-ui channel](https://discord.com/channels/786421998426521600/787254601656827924).

This will save you a lot of potentially wasted effort.

rohnsha0 commented:
![image](https://user-images.githubusercontent.com/86848116/186330766-2c08b9d2-77d9-49f7-a88e-074e18587e1c.png)
@stevepiercy I've added some buttons... and changed the buttons appearance too...

rohnsha0 commented:
but if you say I'll limit the changes...
and just push the changed `appearance` and your suggested `div class`

mauritsvanrees reviewed (commented):
The buttons that are missing, have been removed explicitly because they are not supported anymore in Plone. This was part of a big refactoring and simplification of the resource registry.

With your fix this already looks better. Thanks!

I think it would look even better if the four buttons were evenly on two rows. And then center the buttons, so they line out with each other. I am thinking of something like this:

       Update   | Deactivate
    Clear Cache |  Download

Is that something you can try?

rohnsha0 commented:
@mauritsvanrees In my opinion, `UPDATE`, `DEACTIVATE`, `CLEAR CACHE`, `DOWNLOAD` in a single row would look much nicer...

mauritsvanrees commented:
If that fits, that would be fine too.

rohnsha0 commented:
> If that fits, that would be fine too.

![image](https://user-images.githubusercontent.com/86848116/186410944-726776dd-5b29-4cfd-9846-3b52db4908e7.png)
I hope this is good!

rohnsha0 commented:
![image](https://user-images.githubusercontent.com/86848116/186411130-c573e320-fdde-4668-83eb-d45df9845692.png)
Full screen

rohnsha0 commented:
![image](https://user-images.githubusercontent.com/86848116/186411298-0774e177-292a-47c7-9312-a99ac7eb3347.png)

stevepiercy commented:
You did not implement the change as I directed in https://github.com/plone/Products.CMFPlone/issues/3595#issuecomment-1219026687. See [line 100](https://github.com/plone/plone.app.theming/blob/master/src/plone/app/theming/browser/controlpanel.pt#L100)

rohnsha0 commented:
> You did not implement the change as I directed in [plone/Products.CMFPlone#3595 (comment)

Actually @stevepiercy , I haven't pushed anything since our last convo...
But were discussing about changes in the buttons

stevepiercy commented on src/plone/app/theming/browser/controlpanel.pt:
```diff
@@ -181,7 +181,7 @@
                             <form method="get" target="_blank" tal:attributes="action string:${context/absolute_url}/++theme++${theme/name}/@@download-zip">
                                 <button i18n:translate=""
-                                    class="btn btn-primary">Download</button>
+                                    class="btn btn-primary mt-1 px-3">Download</button>
```
A class should not be applied to a single button that resides in a div that contains a group of buttons. Instead the [parent div of the group of buttons](https://github.com/plone/plone.app.theming/pull/213/files#diff-c96bcee60e1865b90e3c96e35ed68375a88a71130e599b8ec757dec76a5f0634R151) should have a class applied, specifically `d-grid gap-1`. See https://getbootstrap.com/docs/5.2/utilities/spacing/#gap.

```suggestion
                                    class="btn btn-primary">Download</button>
```

rohnsha0 replied:
actually, @stevepiercy I've done some slight changes to the block placement and don't need grid spacings
![image](https://user-images.githubusercontent.com/86848116/186424159-8b41a97c-df3e-4934-b1b3-f5f6708860c8.png)

stevepiercy replied:
Those changes do not appear in this PR. Please commit and push the changes.

rohnsha0 replied:
@stevepiercy @mauritsvanrees  Committed changes... kindly have a look!

stevepiercy commented on src/plone/app/theming/browser/controlpanel.pt:
```diff
@@ -97,7 +97,7 @@
         </p>
 
-        <div class="btn-group">
+        <div class="m-1">
```
Thank you. This is correct.

rohnsha0 commented:
@jenkins-plone-org please run jobs

ale-rt commented on news/212.bugfix:
```diff
@@ -0,0 +1,2 @@
+Fixed/ Revamped cluttered action buttons in Theming Control Panel
```
```suggestion
The action buttons in the theming control panel have been improved
```

rohnsha0 replied:
@ale-rt sorry' didn't noticed this one ..
I'll edit this one directly! 😑

ale-rt reviewed (approved):
For me this is a nice improvement:
![image](https://user-images.githubusercontent.com/1300763/186504699-a24767c6-8311-4937-9482-387323ffc7e1.png)
I am +1 merging this.

If later we want to improve/refine there is always time.

Thanks a lot for your contribution :)
